### PR TITLE
Fix Common.Tests.GetPrettyName_CannotRead_ReturnsNull test for root user

### PR DESCRIPTION
**Pull request by sirntar**
That test was meant to change privileges of file to `0` and then try to read file, which should resolve in System.Exception, but instead it reads file normally.

I modified test, so it will just check whether file permissions has been set to `None` if user have root permissions, because in that case kernel would ignore privileges check, so the test would fail.

Of course there is an option to change code in corelib instead of test, but I'm not sure if root should be artificially restricted.

Part of https://github.com/dotnet/runtime/issues/84834
cc @tomeksowi @wscho77 @HJLeee @JongHeonChoi @t-mustafin @gbalykov @clamp03 @yurai007 @jkotas

### CODE REVIEW

**Comment by am11** [comment]
(Change is not specific to RISC-V)

cc @wfurt

File: src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs [hunk 1]
```diff
@@ -52,7 +52,17 @@ public void GetPrettyName_CannotRead_ReturnsNull()
...
             File.SetUnixFileMode(path, UnixFileMode.None);
-            Assert.ThrowsAny<Exception>(() => File.ReadAllText(path));
+
+            // If user have root permissions, kernel doesn't care about access priviliges,
+            // so there is no point in expecting System.Exception
+            if (!Environment.IsPrivilegedProcess)
+            {
+                Assert.ThrowsAny<Exception>(() => File.ReadAllText(path)); 
+            }
+            else
```
Comment (wfurt) [inline src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs:62]: I would probably make the test conditional and if needed add separate set of tests for privilege access. I would expect more tests that checks permissions.

But in general I think is is ok if just want to get unblocked to run tests as root.

Comment (Sirntar) [reply src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs:62]: I wasn't certain whether I should modify the test itself or the core library functionality, so this solution was more of a quick fix.

I like your approach of creating a separate set of tests specifically for privileged users. Do you think it's worth implementing in this case? If so, I can start working on that solution tomorrow.

Comment (wfurt) [reply src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs:62]: perhaps @dotnet/area-system-io  can provide more guidance. I personally feel the quick fix + follow-up would be best.

File: src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs [hunk 2]
```diff
@@ -52,7 +52,17 @@ public void GetPrettyName_CannotRead_ReturnsNull()
         {
             string path = CreateTestFile();
             File.SetUnixFileMode(path, UnixFileMode.None);
-            Assert.ThrowsAny<Exception>(() => File.ReadAllText(path));
+
+            // If user have root permissions, kernel doesn't care about access priviliges,
```
Comment (danmoseley) [inline src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs:56]:
```suggestion
            // If user have root permissions, kernel doesn't care about access privileges,
```

Comment (danmoseley) [reply src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs]: (nit, just merged in case that saves you time)

**Comment by Sirntar** [comment]
@dotnet-policy-service agree company="Samsung"

File: src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs [hunk 3]
```diff
@@ -50,9 +50,38 @@ public void GetPrettyName_NoFile_ReturnsNull()
...
+
+            string? name = Interop.OSReleaseFile.GetPrettyName(path);
+            Assert.Null(name);
+        }
+
+        [Fact, PlatformSpecific(TestPlatforms.Linux)]
+        public void GetPrettyName_NonePrivileges_CanRead_ReturnsNull()
+        {
+            // This test only applies for root users.
+            if (!Environment.IsPrivilegedProcess)
```
Comment (tomeksowi) [inline src/libraries/Common/tests/Tests/Interop/OSReleaseTests.cs:72]: Perhaps put `[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsPrivilegedProcess))]` on the test signatures? (I think it works with `Environment` too).

**Review by tomeksowi — APPROVED** [review]

**Review by wfurt — APPROVED** [review]
LGTM

**Pull request merged**